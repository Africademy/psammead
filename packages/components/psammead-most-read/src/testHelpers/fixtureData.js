export const itemsLTR = [
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];

export const itemsRTL = [
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];

export const links = [
  {
    language: 'news',
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    language: 'arabic',
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];
