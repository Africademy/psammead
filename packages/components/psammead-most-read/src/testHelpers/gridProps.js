const mostReadListGridProps = maxTwoColumns => ({
  enableGelGutters: true,
  enableGelMargins: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 8,
    group5: maxTwoColumns ? 8 : 20,
  },
});

const mostReadItemGridProps = {
  item: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 3,
    group4: 4,
    group5: 4,
  },
};

export { mostReadListGridProps, mostReadItemGridProps };
