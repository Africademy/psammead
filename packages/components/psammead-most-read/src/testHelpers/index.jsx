import React from 'react';
import { shape, string, oneOf, number } from 'prop-types';
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import { scriptPropType } from '@bbc/gel-foundations/prop-types';
import Timestamp from '@bbc/psammead-timestamp';
import {
  MostReadTitle,
  MostReadList,
  MostReadItemWrapper,
  MostReadLink,
  MostReadRank,
} from '..';

const lastUpdated = (script, service) => (
  // This will return the provided english translations
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

export const getItem = (service, withTimestamp = false) => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  const timestamp = withTimestamp ? lastUpdated(latin, service) : null;
  return {
    id: Math.floor(Math.random() * 100000) + 1,
    title: text,
    href: `${baseUrl}${articlePath}`,
    timestamp,
  };
};

export const getItemWrapperArray = ({
  numberOfItems,
  service,
  script,
  dir,
}) => {
  const itemWrapperArray = [];

  for (let i = 1; i <= numberOfItems; i += 1) {
    itemWrapperArray.push(
      <MostReadItemWrapper dir={dir} key={i}>
        <MostReadRank
          service={service}
          script={script}
          listIndex={i}
          numberOfItems={numberOfItems}
          dir={dir}
        />
        <MostReadLink
          dir={dir}
          href={getItem(service).href}
          service={service}
          script={script}
          title={getItem(service).title}
        />
      </MostReadItemWrapper>,
    );
  }
  return itemWrapperArray;
};

export const renderMostRead = ({
  service,
  script,
  dir,
  header,
  numberOfItems,
}) => (
  <>
    <MostReadTitle
      header={header}
      service={service}
      script={script}
      dir={dir}
    />
    <MostReadList numberOfItems={numberOfItems} dir={dir}>
      {getItemWrapperArray({
        numberOfItems,
        service,
        script,
        dir,
      }).map(item => item)}
    </MostReadList>
  </>
);

renderMostRead.propTypes = {
  service: string.isRequired,
  script: shape(scriptPropType).isRequired,
  dir: oneOf(['rtl', 'ltr']),
  header: string.isRequired,
  numberOfItems: number.isRequired,
};

renderMostRead.defaultProps = {
  dir: 'ltr',
};
