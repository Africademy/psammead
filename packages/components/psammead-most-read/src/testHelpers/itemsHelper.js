import React from 'react';
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { shape, string, number } from 'prop-types';
import Timestamp from '@bbc/psammead-timestamp';

const lastUpdated = (script, service) => (
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

export const getItem = service => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    id: Math.random(),
    title: text,
    href: `${baseUrl}${articlePath}`,
  };
};

export const getItemWithTimestamp = (script, service) => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    id: Math.random(),
    title: text,
    href: `${baseUrl}${articlePath}`,
    timestamp: lastUpdated(script, service),
  };
};

export const getItems = (service = 'news', number) =>
  Array(number).fill(getItem(service));

export const itemPropTypes = shape({
  id: number,
  title: string,
  href: string,
  timestamp: number,
});
