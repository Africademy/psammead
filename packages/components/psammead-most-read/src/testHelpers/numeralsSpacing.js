const notBengali = {
  minWidth: '3rem',

}