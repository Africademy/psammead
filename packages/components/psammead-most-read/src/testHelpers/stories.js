import { withServicesKnob } from '@bbc/psammead-storybook-helpers';

export const arabicServiceDecorator = withServicesKnob({
  defaultService: 'arabic',
});

export const bengaliServiceDecorator = withServicesKnob({
  defaultService: 'bengali',
});

export const burmeseServiceDecorator = withServicesKnob({
  defaultService: 'burmese',
});

export const newsServiceDecorator = withServicesKnob({
  defaultService: 'news',
});
