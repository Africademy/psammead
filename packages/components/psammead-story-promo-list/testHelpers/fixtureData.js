const storyPromoData = [
  {
    image: {
      alt: 'Thanos',
      src:
        'https://www.log.com.tr/wp-content/uploads/2018/08/avengers-infinity.jpg',
    },
    info: {
      headline: 'Thanos',
      summary: 'Josh Brolin',
      time: '2019-03-01T14:00+00:00',
    },
  },
  {
    image: {
      alt: 'Iron Man',
      src: 'https://www.log.com.tr/wp-content/uploads/2019/02/iron-man.jpg',
    },
    info: {
      headline: 'Iron Man',
      summary: 'Robert Downey',
      time: '2019-03-01T14:00+00:00',
    },
  },
  {
    image: {
      alt: 'Captain American',
      src:
        'https://znews-photo.zadn.vn/w660/Uploaded/rotntt/2013_10_17/chris.jpg',
    },
    info: {
      headline: 'Captain American',
      summary: 'Chris Evans',
      time: '2019-03-01T14:00+00:00',
    },
  },
  {
    image: {
      alt: 'Thor',
      src:
        'https://multimedia.larepublica.pe/660x392/larepublica/imagen/2019/04/29/noticia-1556577358-portadatres.png',
    },
    info: {
      headline: 'Thor',
      summary: 'Chris Hemsworth',
      time: '2019-03-01T14:00+00:00',
    },
  },
  {
    image: {
      alt: 'Captain Marvel',
      src:
        'https://cms-static.wehaacdn.com/illinoistimes-com/images/film.25327.widea.0.jpg',
    },
    info: {
      headline: 'Captain Marvel',
      summary: 'Brie Larson',
      time: '2019-03-01T14:00+00:00',
    },
  },
];

export default storyPromoData;
