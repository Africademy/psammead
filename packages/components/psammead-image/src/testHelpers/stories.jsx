import React from 'react';
import { storiesOf } from '@storybook/react';
import Readme from '../../README.md';
import { custom, landscape, portrait, square } from './fixtureData';

function getProps(image, includeHeight) {
  const props = {
    alt: image.alt,
    src: image.src,
    srcset: image.srcset,
    width: image.width,
  };
  props.height = includeHeight ? image.height : null;

  return props;
}

const stories = (
  Component,
  title,
  includeHeight = false,
  additionalProps = {},
  styleDecorator = storyFn => storyFn(),
) =>
  storiesOf(title, module)
      .addDecorator(styleDecorator)
    .add('landscape image', () => (
      <Component {...getProps(landscape, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('portrait image', () => (
      <Component {...getProps(portrait, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('square image', () => (
      <Component {...getProps(square, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('custom ratio image', () => (
      <Component {...getProps(custom, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('image with srcset', () => (
      <Component
        {...getProps(landscape, includeHeight)}
        srcset={landscape.srcset}
        {...additionalProps}
      />
    ), { notes: Readme });

export default stories;
