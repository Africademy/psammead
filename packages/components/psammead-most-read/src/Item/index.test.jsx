import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import Timestamp from '@bbc/psammead-timestamp';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { MostReadLink, MostReadRank } from '.';
import { items, itemsRTL } from '../testHelpers/fixtureData';

const lastUpdated = (script, service) => (
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

describe('MostReadLink', () => {
  shouldMatchSnapshot(
    'should render ltr correctly',
    <MostReadLink link={items[0]} service="news" script={latin} dir="ltr" />,
  );

  shouldMatchSnapshot(
    'should render rtl correctly',
    <MostReadLink
      link={itemsRTL[0]}
      service="persian"
      script={arabic}
      dir="rtl"
    />,
  );

  shouldMatchSnapshot(
    'should render with last updated date correctly',
    <MostReadLink
      link={items[0]}
      lastUpdated={lastUpdated(latin, 'news')}
      service="news"
      script={latin}
      dir="ltr"
    />,
  );
});

describe('MostReadRank', () => {
  shouldMatchSnapshot(
    'should render ltr correctly',
    <MostReadRank service="news" script={latin}>
      5
    </MostReadRank>,
  );

  shouldMatchSnapshot(
    'should render ltr with double digits correctly',
    <MostReadRank service="news" script={latin}>
      10
    </MostReadRank>,
  );

  shouldMatchSnapshot(
    'should render rtl correctly',
    <MostReadRank service="persian" script={arabic}>
      Û²
    </MostReadRank>,
  );
});
