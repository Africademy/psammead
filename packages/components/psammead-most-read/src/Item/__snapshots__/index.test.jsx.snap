// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostReadItemWrapper should render ltr correctly with 10 items 1`] = `
<div>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 keUhmJ"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          1
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 dYgNNo"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          2
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 keUhmJ"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          3
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 dYgNNo"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          4
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 bOhhdK"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          5
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 iAohyw"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          6
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 efLOFq"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          7
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 iAohyw"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          8
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 efLOFq"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          9
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="ltr"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 dvvbaz"
        dir="ltr"
      >
        <span
          class="StyledSpan-fs4y6m-3 htomqN"
        >
          10
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 kYQuFW"
        dir="ltr"
      >
        <a
          class="StyledLink-u0d8kd-0 bBFoEo"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </div>
  </li>
</div>
`;

exports[`MostReadItemWrapper should render rtl correctly with 10 items 1`] = `
<div>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 fUznEE"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۱
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 EihbL"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۲
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 fUznEE"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۳
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 EihbL"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۴
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 cgEA"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۵
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 ndXtY"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۶
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 jEYThi"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۷
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 ndXtY"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۸
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 jEYThi"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۹
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
  <li
    class="GridComponent-nf79gm-0 bvkVj StyledGrid-u0d8kd-4 iLbAiY"
    dir="rtl"
    role="listitem"
  >
    <div
      class="ItemWrapper-u0d8kd-3 juBlKT"
    >
      <div
        class="OneColumnWrapper-fs4y6m-0 TwoColumnWrapper-fs4y6m-1 MultiColumnWrapper-fs4y6m-2 jmGdsg"
        dir="rtl"
      >
        <span
          class="StyledSpan-fs4y6m-3 bvXRlE"
        >
          ۱۰
        </span>
      </div>
      <div
        class="StyledItem-u0d8kd-1 fkeZwt"
        dir="rtl"
      >
        <a
          class="StyledLink-u0d8kd-0 OTgcR"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </div>
  </li>
</div>
`;

exports[`MostReadLink should render ltr correctly 1`] = `
<div>
  <div
    class="StyledItem-u0d8kd-1 kYQuFW"
    dir="ltr"
  >
    <a
      class="StyledLink-u0d8kd-0 bBFoEo"
      href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
    >
      Could a computer ever create better art than a human?
    </a>
  </div>
</div>
`;

exports[`MostReadLink should render rtl correctly 1`] = `
<div>
  <div
    class="StyledItem-u0d8kd-1 fkeZwt"
    dir="rtl"
  >
    <a
      class="StyledLink-u0d8kd-0 OTgcR"
      href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
    >
      لماذا يخجل البعض من اسم قريته في مصر؟
    </a>
  </div>
</div>
`;

exports[`MostReadLink should render with last updated date correctly 1`] = `
<div>
  <div
    class="StyledItem-u0d8kd-1 kYQuFW"
    dir="ltr"
  >
    <a
      class="StyledLink-u0d8kd-0 bBFoEo"
      href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
    >
      Could a computer ever create better art than a human?
    </a>
    <div
      class="TimestampWrapper-u0d8kd-2 hlECaE"
    >
      <time
        class="StyledTimestamp-um718p-0 kIQQTq"
        datetime="2019-03-01T14:00+00:00"
      >
        Last updated: 5th November 2016
      </time>
    </div>
  </div>
</div>
`;
